index.html

```diff
@@ -232,39 +235,393 @@ <h3>Create</h3>
+            <li>
+Check the validity of the input <em>identifier</em>. The <em>scheme</em> MUST be
+the value `did`. The <em>method</em> MUST be the value `key`. The
+<em>version</em> MUST be convertible to a positive integer value. The
+<em>multibase value</em> MUST be a string and begin with the letter `z`. If any
+of these requirements fail, an INVALID_DID error MUST be raised.
+            </li>
+            <li>
+Initialize the <em>signature verification method</em> to the result of passing
+<em>identifier</em>, <em>multibase value</em>, and <em>options</em> to
+<a href="#signature-method-creation-algorithm"></a>.
+            </li>
+            <li>
+Initialize the <em>encryption verification method</em> to the result of passing
+<em>identifier</em>, <em>multibase value</em>, and <em>options</em> to
+<a href="#encryption-method-creation-algorithm"></a>.
+            </li>
+            <li>
+Set the `id` property in <em>document</em> to <em>identifier</em>.
+            </li>
+            <li>
+Initialize the `verificationMethod` property in <em>document</em> to an array
+where the first value is the <em>signature verification method</em>.
+            </li>
+            <li>
+Initialize the `authentication`, `assertionMethod`, `capabilityInvocation`,
+and the `capabilityDelegation` property in <em>document</em> to an array
+where the first item is the value of the `id` property in
+<em>signature verification method</em>.
+            </li>
+            <li>
+If <em>options</em> `disableKeyAgreement` is not set to `true`,
+add the <em>encryption verification method</em> value to the
+`verificationMethod` array and initialize the `keyAgreement` property in
+<em>document</em> to an array where the first item is the value of the `id`
+property in <em>encryption verification method</em>.
+            </li>
+            <li>
+Return <em>document</em>.
+            </li>
+          </ol>
+          <p>
+Any alternative algorithm can be used that produces an equivalent result to the
+output from the algorithm above.
+          </p>
+        </section>
+
+        <section class="normative">
+          <h4>Signature Method Creation Algorithm</h4>
+
+          <p>
+The following algorithm can be used for decoding a multibase-encoded multicodec
+value into a verification method that is suitable for verifying digital
+signatures. An <em>identifier</em>, <em>multibase value</em>, and
+<em>options</em> are the required inputs. A <em>verification method</em> is
+the resulting output.
+          </p>
+
+          <ol class="algorithm">
+            <li>
+Initialize <em>verification method</em> to an empty object.
+            </li>
+            <li>
+Set <em>multicodec value</em> and <em>raw public key bytes</em> to the result
+of passing <em>multibase value</em> and <em>options</em> to
+<a href="#decode-public-key-algorithm"></a>.
+            </li>
+            <li>
+Ensure the proper key length of <em>raw public key bytes</em> based on the
```

Comment:
It's safer to validate the point actually exists on the curve rather than assuming that the correct byte length is all that's necessary to validate the public key. I'd suggest utilzing a point validation check here rather than a byte length check.

Reply:
Yes, agreed @kdenhartog. My only concern is how easy it is to do this check across all public key types that are supported -- for example, does libopenssl key import do the check for all these types of public keys: RSA, Ed25519, X25519, secpr1, secpk1... what does this look like for BLS12381, etc?

I guess the test suite could test with known bad public key values for each public key type... I'm going to try to play with some text to add in a "correctness" test and see where I get to.

Reply:
Implemented in c981fab6e5769d23c8f936b641bd89725b4bcf53.

Reply:
That makes sense to me. As for point validation I know elliptic does it (because I caught an invalid curve attack in their ECDH implementation because they didn't perform the check even though they implemented it) and I just checked Openssl [has it](https://github.com/openssl/openssl/blob/master/crypto/ec/ec_key.c#L472). The different kinds of checks are listed in [Section 5.6.2.3 of NIST SP 800-56A Rev. 3](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/nist.sp.800-56Ar3.pdf)